hack/e2e/ensure_kind.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -eux
+
+USR_LOCAL_BIN="/usr/local/bin"
+MINIMUM_KIND_VERSION=v0.26.0
```

Comment:
We have kind 0.23.0 in go.mod, but here we want 0.26.0, is that on purpose?

Reply:
That is a mistake. I didn't pay enough attention when cherry-picking. Now that I think about it I am not sure we need to even install kind. :thinking: 
With minikube we created the cluster before the test started. Now we create it as part of the test suite itself, which makes this script redundant as far as I can tell. I can push a PR on main to see if we can remove it.

Reply:
Checking here: https://github.com/metal3-io/baremetal-operator/pull/2277

Reply:
I added cherry-pick to 0.9 too, then with this we have all branches covered.